### PR TITLE
[TIMOB-23887]:Attributed string link on a TextArea doesn't fire link event

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-23887

### CODE REVIEW

File: iphone/Classes/TiUITextArea.m [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * Please see the LICENSE included with this distribution for details.
  */
 #ifdef USE_TI_UITEXTAREA
-
+#import "TiUIAttributedStringProxy.h"
```
Comment (hansemannn) [inline iphone/Classes/TiUITextArea.m]: Everything related to other proxies should be wrapped:
```objc
#ifdef USE_TI_UIATTRIBUTEDSTRING
#import "TiUIAttributedStringProxy.h"
#endif
```
See the TiUITextWidget.m implementation for details.

File: iphone/Classes/TiUITextArea.m [hunk 2]
```diff
@@ -121,6 +125,88 @@ -(void)adjustOffsetIfRequired:(UITextView*)tv
     }
 }
 
+-(void)checkLinkForTouch:(UITouch *)touch
+{
+    // TIMOB-23887: This is similar to UILabel implementation, rather native link recognizer of UITextView, to support TIMOB-19165.
+    BOOL testLink = (textWidgetView != nil) &&([(TiViewProxy*)[self proxy] _hasListeners:@"link" checkParent:NO]);
```
Comment (hansemannn) [inline iphone/Classes/TiUITextArea.m]: Missing space in `) &&(`.

File: iphone/Classes/TiUITextArea.m [hunk 3]
```diff
@@ -121,6 +125,88 @@ -(void)adjustOffsetIfRequired:(UITextView*)tv
...
+    BOOL isLinkEnabled = [TiUtils boolValue:[[self proxy] valueForUndefinedKey:@"handleLinks"] def:YES];
+    if (testLink && isLinkEnabled && !isEditable)
+    {
+        UITextView* textView = (UITextView*)[self textWidgetView];
+        CGPoint tapPoint = [touch locationInView:textView];
+        [self checkLinkAttributeForString:[textView attributedText] atPoint:tapPoint];
+    }
+}
+
+- (NSUInteger)characterIndexAtPoint:(NSAttributedString*)theString atPoint:(CGPoint)point
```
Comment (hansemannn) [inline iphone/Classes/TiUITextArea.m]: `NSAttributedString*` `NSAttributedString *`

**Review by hansemannn — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiUITextArea.m (hunk 1 above)
Comment (vijaysingh-axway) [inline iphone/Classes/TiUITextArea.m]: Done. Thanks.

**Review by hansemannn — APPROVED** [review]

**Review by ewieberappc — APPROVED** [review]
FR Passed, using:
MacOS 10.12 (16A323)
Studio 4.9.0.201704061825
Ti SDK 6.1.0.v20170503211832
Appc NPM 4.2.9
Appc CLI 6.2.0
Alloy 1.9.11
Xcode 8.3.2 (8E2002)

Attributed strings, inside non-editable text areas, with links now correctly fire the link event when clicked. Tested using the sample code and modified it to fully apply to the test case.

**Pull request merged**